### ISSUE TITLE
Implement local h in NLM denoising

### ISSUE THREAD

**Issue by TimZaragori** [scikit-image/scikit-image#6299]
In NLM denoising, the parameter `h` controls the smoothness in the final image. While skimage implementation requires a scalar for `h` several articles chose to adapt it locally based on local variance ([Dutta et al.](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0081390), [Li et al.](https://aapm.onlinelibrary.wiley.com/doi/full/10.1118/1.4851635)) or intensities ([Qi et al.](https://ieeexplore.ieee.org/document/7582060)). This allows to more flexibility for this parameter depending on the direct neighborhood of each voxel.

Implementation : `h` would be an array with the same shape as `image.shape` or as `image.shape[:-1]` if `multichannel=True`. During computation the same index as `image` would be used to get the current `h` value

A similar modification can be done for the `sigma` parameter to account for local noise.

**Comment by grlee77** [comment]
Yes, this would be a nice feature. There is some memory overhead to having extra images for `h` and/or `sigma`, but I doubt we would want to maintain fully separate implementations for scalar vs. array properties (we already have enough duplicates across the separate 2D/3D/4D cases and fast vs. slow algorithms as is!).

I might have had a branch with an array `sigma` at one point, but would have to look. The 3D non-local means implemented in [DIPY](https://dipy.org) does support a spatially varying sigma.

The other feature that would be nice is axis-specific block size and search window sizes. That is useful for a common case in medical imaging where the slice thickness is substantially larger than the in-plane resolution.

**Comment by ApaarRaina** [comment]
Hi! I'd be interested in working on this. Is this enhancement still something the maintainers would be interested in accepting?
